cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -630,6 +637,11 @@ private V1ObjectMeta createResourceMetadata(Type resourceType, String runnableNa
       .addToLabels(KubeMasterEnvironment.NAMESPACE_PROPERTY, programRuntimeNamespace)
       .addToAnnotations(KubeTwillRunnerService.APP_LABEL, twillSpec.getName())
       .addToAnnotations(KubeTwillRunnerService.START_TIMEOUT_ANNOTATION, Long.toString(startTimeoutMillis));
+    if (kubeTwillConfigs.containsKey(KubeTwillRunnerService.JOB_SKIP_CLEANUP)) {
+      objectMetaBuilder.addToAnnotations(KubeTwillRunnerService.JOB_SKIP_CLEANUP,
```

Comment:
I don't see this annotation being read anywhere in this PR. Is this some special kubernetes annotation?

Reply:
The annotation will be read by KubeTwillController during termination. We are essentially persisting the flag in the annotation instead of KubeTwillController obj. When system worker pods are enabled, appfabric constructs the KubeTwillController retroactively by looking at the k8s Job.